hedera-node/hedera-schedule-service-impl/src/test/java/com/hedera/node/app/service/schedule/impl/test/handlers/ScheduleCreateHandlerTest.java

```diff
@@ -16,144 +16,81 @@
-        given(schedulerAccount.key()).willReturn(schedulerKey);
-        given(accountStore.getAccountById(payer)).willReturn(payerAccount);
-        given(payerAccount.key()).willReturn(payerKey);
-
-        final var context = new FakePreHandleContext(accountStore, txn);
-
-        subject.preHandle(context);
-
-        basicContextAssertions(context, 1);
-        assertEquals(schedulerKey, context.payerKey());
-        assertEquals(Set.of(adminKey), context.requiredNonPayerKeys());
-
-        final var innerContext = context.innerContext();
-        basicContextAssertions(innerContext, 0);
-        assertEquals(payer, innerContext.payer());
-        assertEquals(payerKey, innerContext.payerKey());
-
-        verify(dispatcher).dispatch(innerContext);
+        realPreContext = new PreHandleContextImpl(
+                mockStoreFactory, scheduleCreateTransaction(payer), testConfig, mockDispatcher);
+        subject.preHandle(realPreContext);
+
+        Assertions.assertEquals(2, realPreContext.requiredNonPayerKeys().size());
+        Assertions.assertEquals(1, realPreContext.optionalNonPayerKeys().size());
+        Assertions.assertEquals(schedulerKey, realPreContext.payerKey());
+        Assertions.assertEquals(Set.of(payerKey, adminKey), realPreContext.requiredNonPayerKeys());
+        Assertions.assertEquals(Set.of(payerKey), realPreContext.optionalNonPayerKeys());
     }
 
     @Test
     void preHandleScheduleCreateVanillaNoAdmin() throws PreCheckException {
-        final var subject = new ScheduleCreateHandler(dispatcher);
-        final var txn = scheduleCreateTxnWith(
+        final TransactionBody transactionToTest = ScheduledTxnFactory.scheduleCreateTxnWith(
                 null, "", payer, scheduler, Timestamp.newBuilder().seconds(1L).build());
+        realPreContext = new PreHandleContextImpl(mockStoreFactory, transactionToTest, testConfig, mockDispatcher);
+        subject.preHandle(realPreContext);
 
-        given(accountStore.getAccountById(scheduler)).willReturn(schedulerAccount);
-        given(schedulerAccount.key()).willReturn(schedulerKey);
-        given(accountStore.getAccountById(payer)).willReturn(payerAccount);
-        given(payerAccount.key()).willReturn(payerKey);
-
-        final var context = new FakePreHandleContext(accountStore, txn);
-
-        subject.preHandle(context);
-
-        basicContextAssertions(context, 0);
-        assertEquals(schedulerKey, context.payerKey());
-        assertEquals(Set.of(), context.requiredNonPayerKeys());
-
-        final var innerContext = context.innerContext();
-        basicContextAssertions(innerContext, 0);
-        assertEquals(payer, innerContext.payer());
-        assertEquals(payerKey, innerContext.payerKey());
-
-        verify(dispatcher).dispatch(innerContext);
+        Assertions.assertEquals(1, realPreContext.requiredNonPayerKeys().size());
+        Assertions.assertEquals(1, realPreContext.optionalNonPayerKeys().size());
```

Comment:
Neither of these assertions are needed if we're asserting set equality below

Reply:
Having the number assertion provides an easier error to read (more obvious the counts do not match).  Saved me a lot of time debugging the initial code pass.